ladybug_geometry/geometry3d/face.py

```diff
@@ -1151,6 +1151,108 @@ def sub_faces_by_ratio_sub_rectangle(self, ratio, sub_rect_height, sill_height,
             sub_faces.extend(face.sub_faces_by_ratio(ratio))
         return sub_faces
 
+    def _sub_faces_by_dimensions(self, base_plane, rect_base, rect_height,
+                                 sub_face_height, sub_face_width,
+                                 sill_height, dist_breakup):
+        """Get a list of rectangular Face3D objects using parameters to define them.
+
+        Args:
+            rect_base: A number indicating the length of the base of the
+                parent rectangle.
+            rect_height: A number indicating the length of the height of the
+                parent rectangle.
+            sub_face_height: A number for the target height of the output rectangles.
+                If the ratio is too large for the height, the ratio will take 
+                precedence and the sub-rectangle height will be larger than this value.
+            sub_face_width: A number for the target width of the output rectangles.
+                If the ratio is too large for the width, the ratio will take 
+                precedence and the sub-rectangle width will be larger than this value.
+            dist_breakup: A number for the target breakup separation distance between
+                individual sub-rectangle centerlines.  If this number is larger than
+                the parent rectangle base, only one sub-rectangle will be produced.
+            sill_height: A number for the target height above the bottom edge of
+                the rectangle to start the sub-rectangles. If the ratio is too
+                large for the height, the ratio will take precedence and the
+                sub-rectangle height will be smaller than this value.
+            base_plane: A Plane object in which the rectangle exists.
+                The origin of this plane will be the lower left corner of the
+                rectangle and the X and Y axes will form the sides.
+                Default is the world XY plane.
+
+        Returns:
+            A list of Face3D objects for sub faces.
+        """
+        # if sub_face_height > rect_height, set it to just under rect_height
+        sub_face_height_final = (
+            rect_height - 0.01 if sub_face_height > rect_height - 0.01
+            else sub_face_height)
+        # if sill_height is close to 0, set it to just above 0
+        sill_height_final = (
+            0.01 * rect_height if sill_height < 0.01 * rect_height else sill_height)
+        # adjust sill_height_final if sum of it and sub_face_height_final > rect_height
+        if sub_face_height_final + sill_height_final > rect_height:
+            sill_height_final = rect_height - sub_face_height_final
+
+        max_width_break_up = rect_base / 2
+        num_divisions = (
+            round(rect_base / dist_breakup) if rect_base > dist_breakup / 2 else 1)
+        if (sub_face_width < max_width_break_up and
+            num_divisions * sub_face_width <= rect_base):
```

Comment:
This line should be indented one more or you can just move it to the line above if you can get it under 89 characters.

Reply:
Also, I would not use parentheses here.